components/browser/storage-sync/src/main/java/mozilla/components/browser/storage/sync/PlacesStorage.kt

```diff
@@ -0,0 +1,63 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.browser.storage.sync
+
+import android.content.Context
+import android.support.annotation.VisibleForTesting
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.cancelChildren
+import kotlinx.coroutines.withContext
+import mozilla.appservices.places.PlacesException
+import mozilla.appservices.places.PlacesReaderConnection
+import mozilla.appservices.places.PlacesWriterConnection
+import mozilla.components.concept.sync.AuthInfo
+import mozilla.components.concept.sync.SyncStatus
+
+open class PlacesStorage(context: Context) {
+
+    internal val scope by lazy { CoroutineScope(Dispatchers.IO) }
+    private val storageDir by lazy { context.filesDir }
+
+    @VisibleForTesting
+    internal open val places: Connection by lazy {
+        RustPlacesConnection.init(storageDir)
+        RustPlacesConnection
+    }
+
+    internal val writer: PlacesWriterConnection by lazy { places.writer() }
+    internal val reader: PlacesReaderConnection by lazy { places.reader() }
+
+    /**
+     * Internal database maintenance tasks. Ideally this should be called once a day.
+     */
+    suspend fun runMaintenance() {
```

Comment:
Let's keep `cleanup` and `runMaintanence` at the interface level. You can declare a new concept-storage interface called `Storage` (alongside HistoryStorage and BookmarksStorage), and have these methods live there. Then an abstract `PlacesStorage` will implement `Storage`. 

The reason I think these are better left at the concept level: nothing in the system interacts with a PlacesHistoryStorage, or a PlacesBookmarksStorage. We're always interacting with a _some_ object that implements a HistoryStorage contract, or a BookmarksStorage contract. If `runMaintanence` is hidden within the concrete implementation, we will not be able to actually call these methods from within a `SyncManager`, because it doesn't know about `browser-storage-sync`, it knows about `concept-storage`.

Reply:
That makes sense. Changes made.